Restore `hidden-review-comments-indicator`

chcamiloam opened the pull request:
<!--

Thanks for contributing! 🍄 Do not ignore this template, plz.

1. Does this PR close/fix an existing issue? Write something like `Closes #10`

Help us test and visualize this PR:

2. What pages does this PR affect? Include some REAL URLs where you tested the code
3. If applicable, add demonstrative screenshots or gifs

Lastly:

4. Open the PR as draft and review it yourself first. Fix what you find and explain weird code, if necessary.

-->

- Part of #5897 

## Test URLs
https://github.com/refined-github/sandbox/pull/18/files

## Screenshot
<img width="677" alt="Screen Shot 2022-09-08 at 11 07 41" src="https://user-images.githubusercontent.com/26499470/189083464-cbc57bfc-f690-4970-aa51-ba24b370d979.png">



## Description

Class `.blob-num` was overwriting plugin's css, added `!important` to fix the issue.
Counter was not working, it was using the wrong selector

fregante commented on source/features/hidden-review-comments-indicator.css:
```diff
@@ -49,4 +49,4 @@
 .file-diff-split .rgh-comments-indicator button {
 	padding-right: 1px;
-}
+}
```
Strangely this was not caught by style lint. Every file must end with a line break

fregante commented on source/features/hidden-review-comments-indicator.css:
```diff
@@ -30,7 +30,7 @@
 .rgh-comments-indicator button {
 	position: absolute;
-	right: 100%;
+	right: 90%;
```
Why? It seems too close now

fregante replied:
Maybe it's ok

chcamiloam replied:
The Comments icon is cut in 1800px with `100%` but we can try :)

fregante commented on source/features/hidden-review-comments-indicator.css:
```diff
@@ -8,7 +8,7 @@
 .rgh-comments-indicator:is(::before, ::after) {
 	content: '' !important; /* Needs to override .blob-num’s */
-	position: absolute;
+	position: absolute !important;/* Needs to override .blob-num’s */
```
```suggestion
	position: absolute !important; /* Needs to override .blob-num’s */
```

fregante commented on source/features/hidden-review-comments-indicator.css:
```diff
@@ -8,7 +8,7 @@
 .rgh-comments-indicator:is(::before, ::after) {
 	content: '' !important; /* Needs to override .blob-num’s */
-	position: absolute;
+	position: absolute !important; /* Needs to override .blob-num’s */
```
Is this selector doing anything? I don't see the line added by `border-top` in the attached screenshot

chcamiloam replied:
The class is working, but I'm not so sure about the selector, let me do some testing :)

chcamiloam commented:
Removed the css, I took one by one the changes and if doesn't affect anything (the absolute is still the fix of the main issue).
I tried with `right:100%` and this is the result
<img width="268" alt="Screen Shot 2022-09-08 at 13 11 44" src="https://user-images.githubusercontent.com/26499470/189111701-621082c8-f09c-4533-8be5-49a79e12c9dc.png">

I changed it to 95% this time, it should allow to not have issues with 2digits numbers

chcamiloam commented:
Changed the structure to handle the icon with less css. Everything looks the same but the markup structure is different, add the line again.
<img width="249" alt="Screen Shot 2022-09-08 at 14 46 25" src="https://user-images.githubusercontent.com/26499470/189130253-6300d547-5a20-4989-bd3a-add12aa3c018.png">

chcamiloam commented:
Added new selector to change markup in split-view, this may seem unnecessary but I really wanted the line in bot numbers.
The space in the left is lower in split view, so it's not possible to have the number.

<img width="928" alt="image" src="https://user-images.githubusercontent.com/26499470/189303221-83f0d5ae-8b6e-46de-8058-366ca5128f4d.png">

fregante commented:
We try to keep the amount of code to a minimum. The previous "complex CSS" worked without changes to the DOM. I'd prefer to keep it that way

chcamiloam commented:
I can't see how to draw the line in both number columns without changing the markup, can I get a hand here? this is how it looks right now.

<img width="928" alt="image" src="https://user-images.githubusercontent.com/26499470/189311892-0b496d0d-ccaa-4968-9bbb-ab65a555c244.png">

fregante commented on source/features/hidden-review-comments-indicator.css:
```diff
@@ -1,4 +1,4 @@
-.rgh-comments-indicator {
+:root .rgh-comments-indicator {
```
This was the only fix required to make the button visible again.

The reason why the line isn't being shown is because GitHub now uses `::before` to show the actual line, so our code fighting with GitHub’s own style.

An easier solution would be to either:

- only use ::after, which doesn't seem to be used by GitHub
- ensure that all the CSS actually applies OR figure out why it doesn't

I undid the previous changes because I don’t think they were going in the right direction for the most part. I'd be ok merging this as is for now and the line can be restored later.

fregante commented on source/features/hidden-review-comments-indicator.css:
```diff
@@ -29,7 +22,7 @@
 .rgh-comments-indicator button {
 	position: absolute;
-	right: 95%;
+	left: -40px;
```
`left: -40px` only works for the specific icon size, font size and character count.
`right: 100%` is the correct alignment and it works with any amount of text in this button.

The only problem lies in the button’s content itself, which seems to have excessive padding (which exists to expand its clickable area, but it should be counteracted by negative margins)
